packages/core/src/utils/MessageValidator.ts

```diff
@@ -11,4 +11,8 @@ export class MessageValidator {
   public static validate<T extends object>(classInstance: T) {
     return validateOrReject(classInstance)
   }
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  public static validateSync<T extends object>(classInstance: T) {
+    return validateSync(classInstance)
+  }
```

Comment:
Instead of adding the validation logic to the json transformer, what if we add it to here. This way if you use the validator directly you'll get the same error types as if you were using the json transformer. By handling this at the lowest level (the validator class) we provide more consistency in how validation errors work. This will also keep the json transformer simpler and focused on transformation

Reply:
ahhh yes, excellent idea!